frameworks/kafka/src/main/java/com/mesosphere/sdk/kafka/api/TopicController.java

```diff
@@ -0,0 +1,171 @@
+        this.cmdExecutor = cmdExecutor;
+    }
+
+    @GET
+    public Response topics() {
+        try {
+            return Response.ok(kafkaZkClient.listTopics(), MediaType.APPLICATION_JSON).build();
+        } catch (Exception ex) {
+            log.error("Failed to fetch topics with exception: " + ex);
+            return Response.serverError().build();
+        }
+    }
+
+    @GET
+    @Path("/{name}")
+    public Response getTopic(@PathParam("name") String topicName) {
+        try {
+            return Response.ok(kafkaZkClient.getTopic(topicName), MediaType.APPLICATION_JSON).build();
+        } catch (Exception ex) {
+            log.error("Failed to fetch topic: " + topicName + " with exception: " + ex);
+            return Response.serverError().build();
+        }
+    }
+
+    @POST
+    public Response createTopic(
+            @QueryParam("name") String name,
+            @QueryParam("partitions") String partitionCount,
+            @QueryParam("replication") String replicationFactor) {
+
+        try {
+            int partCount = Integer.parseInt(partitionCount);
+            int replFactor = Integer.parseInt(replicationFactor);
+            JSONObject result = cmdExecutor.createTopic(name, partCount, replFactor);
+            return Response.ok(result.toString(), MediaType.APPLICATION_JSON).build();
+        } catch (Exception ex) {
+            log.error("Failed to create topic: " + name + " with exception: " + ex);
+            return Response.serverError().build();
+        }
+    }
+
+    /* Below is copied from dcos-kafka-service */
+
+    @GET
+    @Path("/unavailable_partitions")
+    public Response unavailablePartitions() {
+        try {
+            JSONObject obj = cmdExecutor.unavailablePartitions();
+            return Response.ok(obj.toString(), MediaType.APPLICATION_JSON).build();
+        } catch (Exception ex) {
+            log.error("Failed to fetch topics with exception: " + ex);
+            return Response.serverError().build();
+        }
+    }
+
+    @GET
+    @Path("/under_replicated_partitions")
+    public Response underReplicatedPartitions() {
+        try {
+            JSONObject obj = cmdExecutor.underReplicatedPartitions();
+            return Response.ok(obj.toString(), MediaType.APPLICATION_JSON).build();
+        } catch (Exception ex) {
+            log.error("Failed to fetch topics with exception: " + ex);
+            return Response.serverError().build();
+        }
+    }
+
+    @PUT
+    @Path("/{name}")
+    public Response operationOnTopic(
+            @PathParam("name") String name,
+            @QueryParam("operation") String operation,
+            @QueryParam("key") String key,
+            @QueryParam("value") String value,
+            @QueryParam("partitions") String partitions,
+            @QueryParam("messages") String messages) {
+
+        try {
+            JSONObject result = null;
+            List<String> cmds = null;
+
+            if (operation == null) {
```

Comment:
If each operation has its own endpoint, the paths will be the same, though anything automated being run by a user somewhere that depends on the error codes being the same would break.